### PR TITLE
docs(crashlytics): Improve Crashlytics documentation around catching sync/async exceptions.

### PR DESCRIPTION
**Pull request by K9i-0**
## Description

This PR updates the documentation to clarify error handling and exception reporting with FirebaseCrashlytics in Flutter applications. The changes aim to provide clearer instructions for developers integrating FirebaseCrashlytics.

- The original text written for [using runZonedGuarded](https://github.com/firebase/flutterfire/pull/9669) was unnatural.
  - [47e7b80](https://github.com/firebase/flutterfire/pull/10772/commits/47e7b800d85759844a78d048bad1a4f67d3e4388)

- The documentation's phrasing has been updated based on the [example code](https://github.com/firebase/flutterfire/blob/f36785770a6d2c3de981d5bd339166a22af617f4/packages/firebase_crashlytics/firebase_crashlytics/example/lib/main.dart#L28-L31).
  - [272a9d2](https://github.com/firebase/flutterfire/pull/10772/commits/272a9d222e78808fc1dbb2afb8c6a383050e5328)

## Related Issues

None. This PR is not related to any existing issues from the issue database.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by russellwheatley — CHANGES_REQUESTED** [review]

File: docs/crashlytics/_customize-crash-reports.md [hunk 1]
```diff
@@ -23,20 +23,16 @@ to disk to be sent along with the next fatal report or when the app restarts.
...
   await Firebase.initializeApp();
-  bool weWantFatalErrorRecording = true;
   FlutterError.onError = (errorDetails) {
-    if(weWantFatalErrorRecording){
-      FirebaseCrashlytics.instance.recordFlutterFatalError(errorDetails);
-    } else {
-      FirebaseCrashlytics.instance.recordFlutterError(errorDetails);
-    }
+    // If you wish to record a "non-fatal" exception, please use `FirebaseCrashlytics.instance.recordFlutterError` instead
+    FirebaseCrashlytics.instance.recordFlutterFatalError(errorDetails);
```
[the remark this answers is not in the document]
Comment (K9i-0) [reply docs/crashlytics/_customize-crash-reports.md]: I have updated it.
[4c74317](https://github.com/firebase/flutterfire/pull/10772/commits/4c74317d2e55c512cafb145c6cd0ae6d55bb729b)

**Comment by russellwheatley** [comment]
@kevinthecheung - we've made some changes to Crashlytics documentation if you could give it a review, that would be great. Thanks.

File: docs/crashlytics/_customize-crash-reports.md [hunk 2]
```diff
@@ -23,20 +23,25 @@ to disk to be sent along with the next fatal report or when the app restarts.
...
 
   await Firebase.initializeApp();
-  bool weWantFatalErrorRecording = true;
   FlutterError.onError = (errorDetails) {
-    if(weWantFatalErrorRecording){
-      FirebaseCrashlytics.instance.recordFlutterFatalError(errorDetails);
-    } else {
-      FirebaseCrashlytics.instance.recordFlutterError(errorDetails);
-    }
+    // If you wish to record a "non-fatal" exception, please use `FirebaseCrashlytics.instance.recordFlutterError` instead
```
Comment (rachelsaunders) [inline docs/crashlytics/_customize-crash-reports.md:34]: Throughout --

1) pls change "wish" to "want"
2) pls remove the word "please" (we don't use "please" in our tech docs)

**Comment by russellwheatley** [comment]
@K9i-0 - do you mind changing the above mentioned by @rachelsaunders & @mrober, please? We can get this PR merged once done. Thanks.

**Comment by russellwheatley** [comment]
Closing in favour of: https://github.com/firebase/flutterfire/pull/11275

**Pull request closed without merging**